### DISCUSSION TITLE
Too wide permissions in diagrams.net for Docs(google)

### DISCUSSION THREAD

**Discussion by alexander-krasilnikov** [jgraph/drawio#3745] [General]
Hello,

just curious why "diagrams.net for Docs" does need access to ALL my documents even those, which are not diagram related(see screenshot):

<img width="865" alt="image" src="https://github.com/jgraph/drawio/assets/139350961/aa0e3935-a2d5-427f-8597-7eac41fe16e6">

Regards,
Alexander

**Comment by davidjgraph** [comment]
The other possible permission is access to goggle drive files that were created by you with the app or have been opened by you into the app from the drive.google.com interface.

The problem with this one is if you wanted to add a diagram created by someone else this would just fail and we wouldn't be able to explain why.

It's not ideal, no.

**Reply by alexander-krasilnikov** [reply]
Exactly - plugin for Drive demands an access only to diagrams created by me with the plugin - not to all of my documents. I'd say it makes pretty difficult to justify usage of such plugin for Docs in enterprise environment(actually in private too).
